both/quizModel.js

```diff
@@ -0,0 +1,197 @@
+Quiz = function(){
+    var quiz = this;
+    this.addNewQuestion = function(val){
+        //check if the question already exists
+        if (quiz.questions == undefined){
+            quiz.questions = [];
+        }
+        for (var q in quiz.questions){
+
+        };
+
+        quiz.questions.push(val);
+    };
+
+};
+
+Quiz.convertToQuizObject = function(object){
+
+    if (object == undefined) return;
+    var quiz = new Quiz();
+
+    quiz._id = object._id;
+    quiz.title = object.title;
+    quiz.lessonID = object.lessonID;
+    quiz.questions = object.questions;
+    quiz.userAttempts = object.userAttempts;
+
+    return quiz;
+};
+
+Quiz.generateQuestion = function(questionType, quizId ){
+    var question = new Object();
+    question.quizId = quizId;
+    question.id = Random.id(); //assign an id to the question - need this for checking and validating answers
+    question.questionType = questionType;
+
+    //if the question type is true-or-false, populate the answer options
+    if (question.questionType == QuizOptions.TRUE_OR_FALSE){
+        question.optionTitles = [];
+        var trueOption = Quiz.generateAnswerOption("True", false, 0);
+
+        question.optionTitles.push(trueOption);
+        var falseOption = Quiz.generateAnswerOption("False", false, 1);
+        question.optionTitles.push(falseOption);
+    };
+
+    question.description = "";
+    question.title = "";
+    question.options = [];
+    return question;
+};
+
+Quiz.generateAnswerOption = function (title, isSelected, index){
+    var option = {};
+    option.title = title;
+    option.isSelected = isSelected;
+    option.index = index;
+
+    return option
+}
+
+
+Object.defineProperty(Quiz, "_title", {
+    get: function title(){
+        return this._title;
+    },
+    set: function title(val){
+        this._title = val;
+    }
+});
+Object.defineProperty(Quiz, "_questions", {
```

Comment:
SimpleSchema handles this automatically. Then, we use plain JavaScript to access object properties.